{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa689e8c-770e-4205-9c5c-aae4eeb48b60",
   "metadata": {},
   "source": [
    "# Simulated Data PhylogicNDT Reviewer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2111dffe-a1a8-491f-9059-304d814d0231",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9450b50f-6b4c-4ef6-b18b-ddf421179540",
   "metadata": {},
   "outputs": [],
   "source": [
    "from PhylogicNDTReviewer.Reviewers.PhylogicNDTReviewer import PhylogicNDTReviewer\n",
    "\n",
    "import pandas as pd\n",
    "from dash import html\n",
    "import dalmatian\n",
    "import pickle\n",
    "import os\n",
    "import sys"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23b7877b-c841-4f57-adfc-4707a7c0f138",
   "metadata": {},
   "source": [
    "# Set up simuated tumor data\n",
    "\n",
    "If `SimulatedTumorData` submodule is not in the `example_notebooks` directory, clone the `SimulatedTumorData` repository in the `example_notebooks` directory\n",
    "```\n",
    "cd example_notebooks\n",
    "git submodle add https://github.com/getzlab/SimulatedTumorData\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90779fc9-619d-4c28-9240-ebbe0f13d620",
   "metadata": {},
   "source": [
    "## Run to install packages for loading the patient and sample data. Only need to run once in your environment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90862ffb-2d5e-46f4-a9de-6a7949b2c94f",
   "metadata": {},
   "outputs": [],
   "source": [
    "%pip install -e SimulatedTumorData/."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21e2a03c-a5db-4e1f-9f06-278b141edf03",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "85c639be-f1f4-4390-a952-ecd471359cec",
   "metadata": {},
   "outputs": [],
   "source": [
    "from SimulatedTumorData.src.generate_simulated_data import load_patients_and_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f4e1430c-42e1-4c53-a181-54095269b355",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/cchu/Desktop/Methods/SimulatedTumorData/sim_data/patient1/phylogicNDT_results_1000\n",
      "loading existing CNV pickle file SimulatedTumorData/sim_data/patient1/patient1.cnv_events.pkl\n",
      "SimulatedTumorData/sim_data/patient1/sample_coverage/p1_t1.binned_coverage.tsv already exists.\n",
      "SimulatedTumorData/sim_data/patient1/sample_coverage/p1_t2.binned_coverage.tsv already exists.\n",
      "SimulatedTumorData/sim_data/patient1/sample_coverage/p1_t3.binned_coverage.tsv already exists.\n",
      "patient variants path exists: SimulatedTumorData/sim_data/patient1/patient1.variants.tsv\n",
      "Sample p1_t1 has variants_fn: SimulatedTumorData/sim_data/patient1/sample_muts/p1_t1.variants.tsv\n",
      "Sample p1_t2 has variants_fn: SimulatedTumorData/sim_data/patient1/sample_muts/p1_t2.variants.tsv\n",
      "Sample p1_t3 has variants_fn: SimulatedTumorData/sim_data/patient1/sample_muts/p1_t3.variants.tsv\n",
      "Run SimulatedTumorData/sim_data/patient1/sample_mut_vcf/p1_t1.variants.vcf through (nexus-snp hg19 RefSeq).\n",
      "Generated sif file: SimulatedTumorData/sim_data/patient1/patient1.phylogicNDT.sif\n",
      "Phylogic results are already loaded here: SimulatedTumorData/sim_data/patient1/phylogicNDT_results_1000\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient1/sample_ABSOLUTE_results/p1_t1/ABSOLUTE_results\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient1/sample_ABSOLUTE_results/p1_t2/ABSOLUTE_results\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient1/sample_ABSOLUTE_results/p1_t3/ABSOLUTE_results\n",
      "Generated cancer_patient_reviewer_sample_data_fn: SimulatedTumorData/sim_data/patient1/cancer_patient_reviewer_data/patient1_samples_data.tsv\n",
      "Generated cancer_patient_reviewer_patient_data_fn: SimulatedTumorData/sim_data/patient1/cancer_patient_reviewer_data/patient1_data.tsv\n",
      "/Users/cchu/Desktop/Methods/SimulatedTumorData/sim_data/patient2/phylogicNDT_results_1000\n",
      "loading existing CNV pickle file SimulatedTumorData/sim_data/patient2/patient2.cnv_events.pkl\n",
      "SimulatedTumorData/sim_data/patient2/sample_coverage/p2_t1.binned_coverage.tsv already exists.\n",
      "SimulatedTumorData/sim_data/patient2/sample_coverage/p2_t2.binned_coverage.tsv already exists.\n",
      "SimulatedTumorData/sim_data/patient2/sample_coverage/p2_t3.binned_coverage.tsv already exists.\n",
      "patient variants path exists: SimulatedTumorData/sim_data/patient2/patient2.variants.tsv\n",
      "Sample p2_t1 has variants_fn: SimulatedTumorData/sim_data/patient2/sample_muts/p2_t1.variants.tsv\n",
      "Sample p2_t2 has variants_fn: SimulatedTumorData/sim_data/patient2/sample_muts/p2_t2.variants.tsv\n",
      "Sample p2_t3 has variants_fn: SimulatedTumorData/sim_data/patient2/sample_muts/p2_t3.variants.tsv\n",
      "Run SimulatedTumorData/sim_data/patient2/sample_mut_vcf/p2_t1.variants.vcf through (nexus-snp hg19 RefSeq).\n",
      "Generated sif file: SimulatedTumorData/sim_data/patient2/patient2.phylogicNDT.sif\n",
      "Phylogic results are already loaded here: SimulatedTumorData/sim_data/patient2/phylogicNDT_results_1000\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient2/sample_ABSOLUTE_results/p2_t1/ABSOLUTE_results\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient2/sample_ABSOLUTE_results/p2_t2/ABSOLUTE_results\n",
      "Not running commands. Outputs in SimulatedTumorData/sim_data/patient2/sample_ABSOLUTE_results/p2_t3/ABSOLUTE_results\n",
      "Generated cancer_patient_reviewer_sample_data_fn: SimulatedTumorData/sim_data/patient2/cancer_patient_reviewer_data/patient2_samples_data.tsv\n",
      "Generated cancer_patient_reviewer_patient_data_fn: SimulatedTumorData/sim_data/patient2/cancer_patient_reviewer_data/patient2_data.tsv\n"
     ]
    }
   ],
   "source": [
    "samples, participants = load_patients_and_samples(\n",
    "    path_to_sim_data='SimulatedTumorData/sim_data'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f4980daa-7110-4908-8491-1aee19354696",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>maf_fn</th>\n",
       "      <th>cluster_ccfs_fn</th>\n",
       "      <th>build_tree_posterior_fn</th>\n",
       "      <th>tumor_molecular_subtype</th>\n",
       "      <th>tumor_morphology</th>\n",
       "      <th>tumor_primary_site</th>\n",
       "      <th>cancer_stage</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>death_date_dfd</th>\n",
       "      <th>follow_up_date</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>gender</th>\n",
       "      <th>notes</th>\n",
       "      <th>treatments_fn</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>participant_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>patient1</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/phylogicN...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/phylogicN...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/phylogicN...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/patient1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient2</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/phylogicN...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/phylogicN...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/phylogicN...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/patient2_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           maf_fn  \\\n",
       "participant_id                                                      \n",
       "patient1        SimulatedTumorData/sim_data/patient1/phylogicN...   \n",
       "patient2        SimulatedTumorData/sim_data/patient2/phylogicN...   \n",
       "\n",
       "                                                  cluster_ccfs_fn  \\\n",
       "participant_id                                                      \n",
       "patient1        SimulatedTumorData/sim_data/patient1/phylogicN...   \n",
       "patient2        SimulatedTumorData/sim_data/patient2/phylogicN...   \n",
       "\n",
       "                                          build_tree_posterior_fn  \\\n",
       "participant_id                                                      \n",
       "patient1        SimulatedTumorData/sim_data/patient1/phylogicN...   \n",
       "patient2        SimulatedTumorData/sim_data/patient2/phylogicN...   \n",
       "\n",
       "               tumor_molecular_subtype tumor_morphology  tumor_primary_site  \\\n",
       "participant_id                                                                \n",
       "patient1                       Unknown          Unknown                 NaN   \n",
       "patient2                       Unknown          Unknown                 NaN   \n",
       "\n",
       "                cancer_stage  vital_status  death_date_dfd  follow_up_date  \\\n",
       "participant_id                                                               \n",
       "patient1                 NaN           NaN             NaN             NaN   \n",
       "patient2                 NaN           NaN             NaN             NaN   \n",
       "\n",
       "                age_at_diagnosis  gender  notes  \\\n",
       "participant_id                                    \n",
       "patient1                      32     NaN    NaN   \n",
       "patient2                      32     NaN    NaN   \n",
       "\n",
       "                                                    treatments_fn  \n",
       "participant_id                                                     \n",
       "patient1        SimulatedTumorData/sim_data/patient1/patient1_...  \n",
       "patient2        SimulatedTumorData/sim_data/patient2/patient2_...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "participants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ad9759fd-eba1-4298-a91c-fe2d14ae7a81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>maf_fn</th>\n",
       "      <th>wxs_purity</th>\n",
       "      <th>collection_date_dfd</th>\n",
       "      <th>cnv_seg_fn</th>\n",
       "      <th>absolute_cnv_seg_fn</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>preservation_method</th>\n",
       "      <th>wxs_ploidy</th>\n",
       "      <th>ABSOLUTE_pp_calls_tab_fn</th>\n",
       "      <th>ABSOLUTE_pp_modes_data_fn</th>\n",
       "      <th>ABSOLUTE_pp_modes_plots_fn</th>\n",
       "      <th>ABSOLUTE_mode_res_rds_fn</th>\n",
       "      <th>ABSOLUTE_mode_tab_fn</th>\n",
       "      <th>ABSOLUTE_plot_fn</th>\n",
       "      <th>ABSOLUTE_SSNV_mode_res_rds_fn</th>\n",
       "      <th>ABSOLUTE_RData</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>p1_t1</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_mu...</td>\n",
       "      <td>0.70</td>\n",
       "      <td>50</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>patient1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.83</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p1_t2</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_mu...</td>\n",
       "      <td>0.45</td>\n",
       "      <td>100</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>patient1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.90</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p1_t3</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_mu...</td>\n",
       "      <td>0.90</td>\n",
       "      <td>120</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_cn...</td>\n",
       "      <td>patient1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.75</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient1/sample_AB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p2_t1</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_mu...</td>\n",
       "      <td>0.50</td>\n",
       "      <td>100</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>patient2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.99</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p2_t2</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_mu...</td>\n",
       "      <td>0.85</td>\n",
       "      <td>150</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>patient2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.99</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p2_t3</th>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_mu...</td>\n",
       "      <td>0.20</td>\n",
       "      <td>180</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_cn...</td>\n",
       "      <td>patient2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.00</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "      <td>SimulatedTumorData/sim_data/patient2/sample_AB...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      maf_fn  wxs_purity  \\\n",
       "sample_id                                                                  \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_mu...        0.70   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_mu...        0.45   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_mu...        0.90   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_mu...        0.50   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_mu...        0.85   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_mu...        0.20   \n",
       "\n",
       "           collection_date_dfd  \\\n",
       "sample_id                        \n",
       "p1_t1                       50   \n",
       "p1_t2                      100   \n",
       "p1_t3                      120   \n",
       "p2_t1                      100   \n",
       "p2_t2                      150   \n",
       "p2_t3                      180   \n",
       "\n",
       "                                                  cnv_seg_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_cn...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_cn...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_cn...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_cn...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_cn...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_cn...   \n",
       "\n",
       "                                         absolute_cnv_seg_fn participant_id  \\\n",
       "sample_id                                                                     \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_cn...       patient1   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_cn...       patient1   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_cn...       patient1   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_cn...       patient2   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_cn...       patient2   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_cn...       patient2   \n",
       "\n",
       "           preservation_method  wxs_ploidy  \\\n",
       "sample_id                                    \n",
       "p1_t1                      NaN        1.83   \n",
       "p1_t2                      NaN        1.90   \n",
       "p1_t3                      NaN        1.75   \n",
       "p2_t1                      NaN        1.99   \n",
       "p2_t2                      NaN        1.99   \n",
       "p2_t3                      NaN        2.00   \n",
       "\n",
       "                                    ABSOLUTE_pp_calls_tab_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                   ABSOLUTE_pp_modes_data_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                  ABSOLUTE_pp_modes_plots_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                    ABSOLUTE_mode_res_rds_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                        ABSOLUTE_mode_tab_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                            ABSOLUTE_plot_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                               ABSOLUTE_SSNV_mode_res_rds_fn  \\\n",
       "sample_id                                                      \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...   \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...   \n",
       "\n",
       "                                              ABSOLUTE_RData  \n",
       "sample_id                                                     \n",
       "p1_t1      SimulatedTumorData/sim_data/patient1/sample_AB...  \n",
       "p1_t2      SimulatedTumorData/sim_data/patient1/sample_AB...  \n",
       "p1_t3      SimulatedTumorData/sim_data/patient1/sample_AB...  \n",
       "p2_t1      SimulatedTumorData/sim_data/patient2/sample_AB...  \n",
       "p2_t2      SimulatedTumorData/sim_data/patient2/sample_AB...  \n",
       "p2_t3      SimulatedTumorData/sim_data/patient2/sample_AB...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8a6b76fb-0a9d-42d1-a728-f0689411716a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/svansete/Code/getzlab/PurityReviewer/example_notebooks/AnnoMate/AnnoMate/ReviewDataInterface.py:38: UserWarning: Loading existing data pkl file\n",
      "  warnings.warn(f\"Loading existing data pkl file\")\n"
     ]
    }
   ],
   "source": [
    "reviewer = PhylogicNDTReviewer()\n",
    "\n",
    "reviewer.set_review_data(\n",
    "    # Enter the file path where you would like your pkl file of your review data to be stored\n",
    "    data_pkl_fn='../phylogicNDT_review.pkl',\n",
    "    # Breifly describe the source of your data and its purpose\n",
    "    description='PhylogicNDT review',\n",
    "    # Reference dataframes and directory from above\n",
    "    participant_df=participants,\n",
    "    sample_df=samples,\n",
    ")\n",
    "\n",
    "# Enter optional driver genes file path (text file with Hugo Symbols in one column)\n",
    "reviewer.set_review_app()\n",
    "reviewer.set_default_review_data_annotations_configuration()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "86dc50bd-4a6e-4fe7-adf7-6d3aaa7d4c55",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setting auto_export_path to ../phylogicNDT_review.auto_export\n",
      "Using ../phylogicNDT_review.auto_export for auto exporting.\n",
      "Dash app running on http://0.0.0.0:8081/\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  1    1.0  Non/Coding\n",
      "coding                      1   14.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  2    1.0  Non/Coding\n",
      "coding                      2   16.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  3    2.0  Non/Coding\n",
      "coding                      3   13.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  4    1.0  Non/Coding\n",
      "coding                      4   10.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  5    4.0  Non/Coding\n",
      "coding                      5    9.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                  6    0.0  Non/Coding\n",
      "coding                      6   15.0  Non/Coding\n",
      "           Cluster_Assignment  count        type\n",
      "annotation                                      \n",
      "non-coding                ALL    9.0  Non/Coding\n",
      "coding                    ALL   77.0  Non/Coding\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      1    2.0  Non/Syn\n",
      "non-synonymous                  1   12.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      2    4.0  Non/Syn\n",
      "non-synonymous                  2   12.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      3    0.0  Non/Syn\n",
      "non-synonymous                  3   13.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      4    1.0  Non/Syn\n",
      "non-synonymous                  4    9.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      5    2.0  Non/Syn\n",
      "non-synonymous                  5    7.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                      6    1.0  Non/Syn\n",
      "non-synonymous                  6   14.0  Non/Syn\n",
      "               Cluster_Assignment  count     type\n",
      "annotation                                       \n",
      "synonymous                    ALL   10.0  Non/Syn\n",
      "non-synonymous                ALL   67.0  Non/Syn\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/svansete/Code/getzlab/PurityReviewer/example_notebooks/AnnoMate/AnnoMate/AppComponents/MutationTableComponent.py:445: SettingWithCopyWarning:\n",
      "\n",
      "\n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "/Users/svansete/Code/getzlab/PurityReviewer/example_notebooks/AnnoMate/AnnoMate/AppComponents/MutationTableComponent.py:446: SettingWithCopyWarning:\n",
      "\n",
      "\n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n"
     ]
    }
   ],
   "source": [
    "reviewer.run(mode='external', port=8081)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b774c894-e840-4d10-a796-6a21eab3ef27",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3fe76d3-15f0-4903-a29d-5ecd1b5c2c23",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
